pkg/gui.py

```
@@ -348,3 +351,98 @@ class SettingsMenu(OtherFrame):
+        self.load_settings()
+
+        # Create a save button
+        self.save_btn = Tk.Button(self, text="Save", command=self.save_settings)
+        self.save_btn.grid(row=4, column=0, columnspan=2, sticky=W+E+N)
+
+    def update_settings(self, settings: dict) -> None:
+        """Update application settings with validation."""
+        with threading.Lock():  # Add a class-level lock
+            tesseract_path = settings.get("tesseract_path")
+            if tesseract_path and not os.path.isfile(tesseract_path):
+                raise ValueError(f"Invalid Tesseract path: {tesseract_path}")
```

Ruff (0.8.0)

Avoid specifying long messages outside the exception class

(TRY003)